atlasdb-config/src/main/java/com/palantir/atlasdb/factory/timelock/TimestampCorroboratingTimelockService.java

```diff
@@ -78,6 +89,21 @@ public StartIdentifiedAtlasDbTransactionResponse startIdentifiedAtlasDbTransacti
         return timestampContainer;
     }
 
+    // pretty weird, I haven't put much effort into this, just to plug the gap really
+    private <T, R> T checkAndUpdateLowerBoundBatch(Supplier<T> timestampContainerSupplier,
+            Function<T, List<R>> responseExtractor,
+            ToLongFunction<R> lowerBoundExtractor,
+            ToLongFunction<R> upperBoundExtractor) {
+        long threadLocalLowerBound = lowerBound.get();
+        T timestampContainers = timestampContainerSupplier.get();
+        List<R> responses = responseExtractor.apply(timestampContainers);
+        responses.forEach(timestampContainer -> {
+            checkTimestamp(threadLocalLowerBound, lowerBoundExtractor.applyAsLong(timestampContainer));
+            updateLowerBound(upperBoundExtractor.applyAsLong(timestampContainer));
```

Comment:
I need to spend a bit of time confirming what these do and determine the most efficient way to do this.